src/metabase/driver/sql_jdbc.clj

```diff
@@ -26,8 +26,12 @@
                (sql.qp/format-honeysql driver honeysql-form)))
 
   ([driver database table honeysql-form]
-   (query driver database (merge {:from [(sql.qp/->honeysql driver (hx/identifier :table (:schema table) (:name table)))]}
-                                 honeysql-form))))
+   (let [table-identifier (binding [hx/*honey-sql-version* (sql.qp/honey-sql-version driver)]
```

Comment:
Could we bind `*honey-sql-version*` in query processing middleware so that it happens in one place? Or would that not cover all of the use cases of these functions?